### PR TITLE
Spending report error fix with some visual changes

### PR DESCRIPTION
**Pull request by carkom**
Fixes a crashing bug that happens when the current day of the month is greater than 28.

### CODE REVIEW

File: packages/desktop-client/src/components/reports/reports/SpendingCard.tsx [hunk 1]
```diff
@@ -60,10 +64,17 @@ export function SpendingCard() {
                   ...styles.mediumText,
                   fontWeight: 500,
                   marginBottom: 5,
+                  color: !difference
+                    ? 'inherit'
+                    : difference <= 0
+                      ? theme.noticeTextLight
```
Comment (twk3) [inline packages/desktop-client/src/components/reports/reports/SpendingCard.tsx:70]: @carkom aren't these colours reversed?  It's currently showing red when this month's difference is a positive

**Review by twk3 — CHANGES_REQUESTED** [review]
@carkom one visual note, otherwise confirmed this fixes this issue regarding day 28 being larger than the zero-indexed 28 day intervalData.

File: packages/desktop-client/src/components/reports/reports/SpendingCard.tsx (hunk 1 above)
Comment (youngcw) [reply packages/desktop-client/src/components/reports/reports/SpendingCard.tsx:70]: Isnt that supposed to mean "you spent more than last month try not to do that again" so a positive difference is bad and negative difference is good?

Comment (carkom) [reply packages/desktop-client/src/components/reports/reports/SpendingCard.tsx:70]: So positive is indicating you've overspent compared to last month.

**Review by youngcw — APPROVED** [review]

File: packages/desktop-client/src/components/reports/reports/SpendingCard.tsx (hunk 1 above)
Comment (twk3) [reply packages/desktop-client/src/components/reports/reports/SpendingCard.tsx:70]: oooh, of course :facepalm:

**Review by twk3 — APPROVED** [review]

**Pull request merged**